SPECS/mozjs/mozjs.spec

```diff
@@ -1,159 +1,307 @@
+rm -rf js/src/octane/
+rm -rf js/src/ctypes/libffi/
+
+popd
+
 %build
-cd js/src
+%if 0%{?build_with_lto}
+# https://github.com/ptomato/mozjs/commit/36bb7982b41e0ef9a65f7174252ab996cd6777bd
+export CARGO_PROFILE_RELEASE_LTO=true
+%endif
+
+# Use bundled autoconf
+export M4=m4
+export AWK=awk
+export AC_MACRODIR=../../build/autoconf/  # modified for AzL
+
+sh ../../build/autoconf/autoconf.sh --localdir=/builddir/build/BUILD/firefox-%{version}/js/src configure.in > configure
+chmod +x configure
 
 %configure \
-    --with-system-icu \
-    --enable-readline \
-    --disable-jemalloc \
+  --with-system-icu \
+  --with-system-zlib \
 %if 0%{?with_check}
-    --enable-tests \
+  --enable-tests \
 %else
-    --disable-tests \
+  --disable-tests \
 %endif
-    --with-system-zlib
+  --disable-strip \
+  --with-intl-api \
+  --enable-readline \
+  --enable-shared-js \
+  --enable-optimize \
+  --disable-debug \
+  --enable-pie \
+  --disable-jemalloc
+
 %make_build
 
 %install
-cd js/src
-
 %make_install
+
+# Fix permissions
 chmod -x %{buildroot}%{_libdir}/pkgconfig/*.pc
-# remove non required files
+
+# Avoid multilib conflicts
+case `uname -i` in
+  i386 | ppc | s390 | sparc )
+    wordsize="32"
+    ;;
+  x86_64 | ppc64 | s390x | sparc64 )
+    wordsize="64"
+    ;;
+  *)
+    wordsize=""
+    ;;
+esac
+
+if test -n "$wordsize"
+then
+  mv %{buildroot}%{_includedir}/mozjs-%{major}/js-config.h \
+     %{buildroot}%{_includedir}/mozjs-%{major}/js-config-$wordsize.h
+
+  cat >%{buildroot}%{_includedir}/mozjs-%{major}/js-config.h <<EOF
+#ifndef JS_CONFIG_H_MULTILIB
+#define JS_CONFIG_H_MULTILIB
+
+#include <bits/wordsize.h>
+
+#if __WORDSIZE == 32
+# include "js-config-32.h"
+#elif __WORDSIZE == 64
+# include "js-config-64.h"
+#else
+# error "unexpected value for __WORDSIZE macro"
+#endif
+
+#endif
+EOF
+
+fi
+
+# Remove unneeded files
+rm %{buildroot}%{_bindir}/js%{major}-config
 rm %{buildroot}%{_libdir}/libjs_static.ajs
-rm -rf %{buildroot}%{_prefix}/src
-find %{buildroot} -type f -name "*.la" -delete -print
 
-# Re-naming library and adding symbolic links to follow typical conventions for libraries.
+# Rename library and create symlinks, following fix-soname.patch
 mv %{buildroot}%{_libdir}/libmozjs-%{major}.so \
    %{buildroot}%{_libdir}/libmozjs-%{major}.so.0.0.0
 ln -s libmozjs-%{major}.so.0.0.0 %{buildroot}%{_libdir}/libmozjs-%{major}.so.0
 ln -s libmozjs-%{major}.so.0 %{buildroot}%{_libdir}/libmozjs-%{major}.so
 
 %check
-cd js/src
-
-TEST_RESULT=0
 # Run SpiderMonkey tests
-PYTHONPATH=tests/lib python3 tests/jstests.py -d -s -t 1800 --no-progress --wpt=disabled ../../js/src/dist/bin/js%{major} || TEST_RESULT=$?
+%if 0%{?require_tests}
+%{python3} tests/jstests.py -d -s -t 2400 --exclude-file=known_failures.txt --no-progress --wpt=disabled ../../js/src/dist/bin/js%{major}
+%else
+%{python3} tests/jstests.py -d -s -t 2400 --exclude-file=known_failures.txt --no-progress --wpt=disabled ../../js/src/dist/bin/js%{major} || :
+%endif
```

Comment:
Unfortunately, this won't work in Azure Linux. At least not yet. The way ptest pass/failure is determined is the exit code of the last command. If we run multiple tests, then to get the failures properly reported you could do something similar to the `%check` step in `edk2.spec`.

Reply:
Pushed an update. See if that's more to your liking.